modules/gdscript/gdscript_parser.cpp

```diff
@@ -7571,6 +7571,10 @@ GDScriptParser::DataType GDScriptParser::_reduce_identifier_type(const DataType
 	}
 
 	if (_get_member_type(base_type, p_identifier, member_type)) {
+		if (current_function && current_function->_static) {
```

Comment:
The problem is that this function is also used to find indexed members, so this will be wrong if that's the case. You can check if `p_base_type` is a `nullptr` because them it'll certainly be looking for members of the current class.

Reply:
done! 👍 
@vnen